### PR TITLE
Update to Gradle 6.0 and fix codenarc warnings

### PR DESCRIPTION
**Pull request by gavra0**
This PR updates Gradle version to 6.0 and also fixes all codenarc warnings.

### CODE REVIEW

File: src/main/groovy/com/google/protobuf/gradle/ExecutableLocator.groovy [hunk 1]
```diff
@@ -76,4 +78,5 @@ class ExecutableLocator implements Named {
   String getPath() {
     return path
   }
+
```
Comment (ejona86) [inline src/main/groovy/com/google/protobuf/gradle/ExecutableLocator.groovy:81]: These style checks by codenarc are a bit obnoxious. Let's disable those rules instead:

```diff
diff --git a/config/codenarc/codenarc.xml b/config/codenarc/codenarc.xml
index f502174..e293eab 100644
--- a/config/codenarc/codenarc.xml
+++ b/config/codenarc/codenarc.xml
@@ -45,6 +45,8 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
   </ruleset-ref>
   <ruleset-ref path='rulesets/exceptions.xml'/>
   <ruleset-ref path='rulesets/formatting.xml'>
+    <exclude name="ClassEndsWithBlankLine"/>
+    <exclude name="ClassStartsWithBlankLine"/>
     <exclude name="Indentation"/>
   </ruleset-ref>
   <ruleset-ref path='rulesets/generic.xml'/>
```

Comment (gavra0) [reply src/main/groovy/com/google/protobuf/gradle/ExecutableLocator.groovy:81]: SGTM, I'll upload a new commit fixing those.

File: src/main/groovy/com/google/protobuf/gradle/ExecutableLocator.groovy [hunk 2]
```diff
@@ -36,6 +37,7 @@ import org.gradle.api.Named
  * configured, the plugin should try to run the executable from system search
  * path.
  */
+@CompileDynamic
```
Comment (voidzcy) [inline src/main/groovy/com/google/protobuf/gradle/ExecutableLocator.groovy:40]: Some classes like this one, are able to be compiled statically. But this PR is just upgrading the compiler, I am fine with just annotating everything as dynamic compile.

Comment (ejona86) [reply src/main/groovy/com/google/protobuf/gradle/ExecutableLocator.groovy:40]: I noticed that as well. But swapping from dynamic to static also changes method resolution. I also decided it is probably better to leave things dynamic, at least for now.

**Pull request merged**